compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp

```diff
@@ -183,6 +183,56 @@ struct ConvertTensorFromElementsPattern
   }
 };
 
+/// Populates given sizes array with the dynamic dims.
+static void sizesForTensor(OpBuilder &builder, SmallVectorImpl<Value> &sizes,
+                           Location loc, ShapedType stp, Value tensor) {
+  for (const auto &d : enumerate(stp.getShape())) {
+    if (d.value() == ShapedType::kDynamic) {
+      Value dim = builder.create<tensor::DimOp>(loc, tensor, d.index());
+      sizes.push_back(dim);
+    }
+  }
+}
```

Comment:
I think we prefer using return values over output parameters: they improve readability, and often provide the same or better performance.

We've been following mixed Google and LLVM style guide. For this part, we follow Google: https://google.github.io/styleguide/cppguide.html#Output_Parameters

For more context: https://abseil.io/tips/176

Reply:
ok i see, changed to return value